hostinfo/sysctl.lua

```diff
@@ -29,39 +29,33 @@ end
 function Info:run(callback)
   if los.type() ~= 'linux' then
     self._error = 'Unsupported OS for sysctl'
-    callback()
-    return
+    return callback()
   end
 
-  local child = spawn('sysctl', {'-A'}, {})
-  local data = ''
-
-  child.stdout:on('data', function(chunk)
-    data = data .. chunk
-  end)
-
-  child:on('exit', function(exit_code)
-    if exit_code ~= 0 then
-      self._error = fmt("sysctl exited with a %d exit_code", exit_code)
-      callback()
-      return
+  local function execCb(err, exitcode, data, stderr_data)
+    if exitcode ~= 0 then
+      self._error = fmt("sysctl exited with a %d exit_code", exitcode)
+      return callback()
     end
     for line in data:gmatch("[^\r\n]+") do
       line = line:gsub("^%s*(.-)%s*$", "%1")
       local _, _, key, value = line:find("([^=^%s]+)%s*=%s*([^=]*)")
       if key ~= nil then
-        local obj = {}
-        obj[key] = value
-        table.insert(self._params, obj)
+        table.insert(self._params, {[key] = value})
       end
     end
-    callback()
-  end)
+    return callback()
+  end
 
-  child:on('error', function(err)
-    self._error = err
-    callback()
+  execFileToBuffers('which', {'sysctl'}, {}, function(err, exitcode, data, stderr)
+    if exitcode ~= 0 then
+      self._error = fmt("sysctl exited with a %d exit_code", exitcode)
+      return callback()
+    end
+    -- Remove /n at the end of data returned from which and call
+    return execFileToBuffers(data:sub(1, data:len()-1), {'-A'}, {}, execCb)
```

Comment:
<img width="416" alt="screen shot 2015-08-04 at 3 33 33 pm" src="https://cloud.githubusercontent.com/assets/3248143/9074408/3a65b624-3abe-11e5-8495-e9918f2bdc19.png">
virgo/util/misc .trim doesnt trim the way i want, over here im trying to remove a /n at the end of the line. 
perhaps i can extend the trim function instead